ocaml/typing/typecore.ml

```diff
@@ -4038,12 +4024,12 @@ let rec approx_type env sty =
   | Some (jty, attrs) -> approx_type_jst env attrs jty
   | None ->
   match sty.ptyp_desc with
-  | Ptyp_arrow (p, ({ ptyp_desc = Ptyp_poly _ } as arg_sty), sty) ->
+  | Ptyp_arrow (p, ({ ptyp_desc = Ptyp_poly _ } as arg_sty), sty, m, _) ->
```

Comment:
`m` is too vague. We can say `arg_mode` and shadow it by `transl_alloc_mode`. Similar to the `p` below.